stats/influxdb/util.go

```diff
@@ -57,3 +58,52 @@ func MakeBatchConfig(conf Config) client.BatchPointsConfig {
 		WriteConsistency: conf.Consistency.String,
 	}
 }
+
+func checkDuplicatedTypeDefinitions(fieldKinds map[string]FieldKind, tag string) error {
+	if _, found := fieldKinds[tag]; found {
+		return errors.Errorf("A tag name (%s) shows up more than once in InfluxDB field type configurations.", tag)
+	}
+	return nil
+}
+
+// MakeFieldKinds reads the Config and returns a lookup map of tag names to
+// the field type their values should be converted to.
+func MakeFieldKinds(conf Config) (map[string]FieldKind, error) {
+	fieldKinds := make(map[string]FieldKind)
+	for _, tag := range conf.TagsAsFields {
+		s := strings.Split(tag, ":")
+		switch len(s) {
+		case 1:
+			err := checkDuplicatedTypeDefinitions(fieldKinds, tag)
+			if err != nil {
+				return nil, err
+			}
+			fieldKinds[tag] = String
+		case 2:
+			fieldName, fieldType := s[0], s[1]
+
+			err := checkDuplicatedTypeDefinitions(fieldKinds, fieldName)
+			if err != nil {
+				return nil, err
+			}
```

Comment:
I think it will be better if we:
1. Either use SplitN with `2` so we never actually have a case where len(s) is bigger and then we will fall in the default with the invalid `type` which is perfectly okay in my opinion
2. Instead of switch here we have an if/else which sets `fieldName` and `fieldType` appropriately and then just continues. This will also enable to inline the checkDuplicatedTypeDefinitions as it will be in only one place.

Reply:
Updated the code following the comments.